pkg/plugin/aggregation/aggregator.go

```diff
@@ -157,13 +163,13 @@ func (a *Aggregator) isComplete() bool {
 	return true
 }
 
-func (a *Aggregator) isResultExpected(result *plugin.Result) bool {
-	_, ok := a.ExpectedResults[result.ExpectedResultID()]
+func (a *Aggregator) isResultExpected(obj keyer) bool {
+	_, ok := a.ExpectedResults[obj.Key()]
 	return ok
 }
 
-func (a *Aggregator) isResultDuplicate(result *plugin.Result) bool {
-	_, ok := a.Results[result.ExpectedResultID()]
+func (a *Aggregator) isResultDuplicate(obj keyer) bool {
```

Comment:
This name is fine as-is since it really is specific to results; I don't care about duplicate status updates. Probably just need to revert these changes on this method; I think I made them just so it would build at some point.